protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/protoc/Protocol.kt

```diff
@@ -328,13 +331,22 @@ private fun toStandard(
                         (ctx.fdp.syntax == "proto3" &&
                             (!fdp.options.hasPacked() ||
                                 (fdp.options.hasPacked() && fdp.options.packed)))),
-            map =
-                fdp.label == LABEL_REPEATED &&
-                    fdp.type == FieldDescriptorProto.Type.TYPE_MESSAGE &&
-                    ctx.findLocal(fdp.typeName).fold(
-                        { false },
-                        { it.options?.mapEntry == true }
-                    ),
+            mapEntry =
+                if (fdp.label == LABEL_REPEATED &&
+                    fdp.type == FieldDescriptorProto.Type.TYPE_MESSAGE
+                ) {
+                    findMapEntry(ctx.fdp, fdp.typeName).filter { it.options.mapEntry }
+                        .fold(
+                            { null },
+                            {
+                                resolveMapEntry(
```

Comment:
Instead of duplicating this search once annotating, convert it to a message here and extract the fields for later use here.